src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -188,7 +188,9 @@ private void UpdateCheckoutTasksAndVariables(IExecutionContext executionContext,
             // RepoClean may be set from the server, so start with the server value
             bool? repoClean = executionContext.Variables.GetBoolean(Constants.Variables.Build.RepoClean);
 
-            foreach (var checkoutTask in steps.Where(x => x.IsCheckoutTask()).Select(x => x as TaskStep))
+            var checkoutTasks = steps.Where(x => x.IsCheckoutTask()).Select(x => x as TaskStep);
```

Comment:
I would add .ToList() on the end since you are immediately calling Count(). Then you can change Count() to Count. And avoid enumerating the list just to count them.